fog/view/server/src/fog_view_service.rs

```diff
@@ -188,4 +197,38 @@ impl<E: ViewEnclaveProxy, DB: RecoveryDb + Send + Sync> FogViewApi for FogViewSe
             send_result(ctx, sink, self.query_impl(request), logger)
         })
     }
+
+    /// Fulfills the query if the MultiViewStoreQueryRequest contains an
+    /// encrypted Query for the store. If it doesn't, then it responds with
+    /// an grpc error that contains the store's hostname.
+    fn multi_view_store_query(
+        &mut self,
+        ctx: RpcContext,
+        request: MultiViewStoreQueryRequest,
+        sink: UnarySink<MultiViewStoreQueryResponse>,
+    ) {
+        mc_common::logger::scoped_global_logger(&rpc_logger(&ctx, &self.logger), |logger| {
+            if let Err(err) = self.authenticator.authenticate_rpc(&ctx) {
+                return send_result(ctx, sink, err.into(), logger);
+            }
+            let mut response = MultiViewStoreQueryResponse::new();
+            for query in request.queries {
+                let result = self.query_impl(query);
+                if let Ok(attested_message) = result {
+                    response.set_query_response(attested_message);
+                    return send_result(ctx, sink, Ok(response), logger);
+                }
+            }
+
+            let mut decryption_error = FogViewStoreDecryptionError::new();
+            let fog_view_store_uri = String::from(self.fog_view_uri.clone().url().clone());
+            decryption_error.set_fog_view_store_uri(fog_view_store_uri);
+            decryption_error.set_error_message(String::from(
+                "Could not decrypt a query embedded in the MultiViewStoreQuery",
+            ));
+            response.set_decryption_error(decryption_error);
```

Comment:
Optional/Equivalent: Use `response.mut_decryption_error()` and `to_string()`
```suggestion
            let decryption_error = response.mut_decryption_error();
            decryption_error.set_fog_view_store_uri(self.fog_view_uri.url().to_string());
            decryption_error.set_error_message(
                "Could not decrypt a query embedded in the MultiViewStoreQuery".to_string()
            );
```

Reply:
Done.